--- mpir-cmake/CMakeLists.txt
+++ mpir-cmake/CMakeLists.txt
@@ -53,7 +53,7 @@ file(GLOB MPQ_SOURCES "mpq/*.c")
 file(GLOB MPZ_SOURCES "mpz/*.c")
 file(GLOB PRINTF_SOURCES "printf/*.c")
 file(GLOB SCANF_SOURCES "scanf/*.c")
-set(SOURCES assert.c compat.c errno.c invalid.c memory.c mp_bpl.c mp_clz_tab.c mp_dv_tab.c mp_get_fns.c mp_minv_tab.c
+set(SOURCES extract-dbl.c assert.c compat.c errno.c invalid.c memory.c mp_bpl.c mp_clz_tab.c mp_dv_tab.c mp_get_fns.c mp_minv_tab.c
 mp_set_fns.c nextprime.c primesieve.c randbui.c randclr.c randdef.c randiset.c randlc2s.c randlc2x.c randmt.c randmts.c
 randmui.c rands.c randsd.c randsdui.c tal-reent.c version.c)