--- zbarlight-1.2/src/zbarlight/__init__.py
+++ zbarlight-1.2/src/zbarlight/__init__.py
@@ -2,12 +2,11 @@ from __future__ import absolute_import, division, print_function, unicode_litera
 import warnings
 
 from PIL import Image
-import pkg_resources
 
 from ._zbarlight import Symbologies, zbar_code_scanner
 
 
-__version__ = pkg_resources.get_distribution('zbarlight').version
+__version__ = "1.2"
 __ALL__ = ['Symbologies', 'UnknownSymbologieError', 'scan_codes', 'qr_code_scanner']